diff --git a/node.gyp b/node.gyp
index 52f421f..ccafd6e 100644
--- a/node.gyp
+++ b/node.gyp
@@ -578,9 +578,6 @@
           '-Wl,-bnoerrmsg',
         ],
       }],
-      ['OS=="linux" and clang==1', {
-        'libraries': ['-latomic'],
-      }],
     ],
   },
 
diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 819dffb..bb3c494 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -1401,14 +1401,6 @@
         ['v8_postmortem_support', {
           'dependencies': ['postmortem-metadata#target'],
         }],
-        # Platforms that don't have Compare-And-Swap (CAS) support need to link atomic library
-        # to implement atomic memory access.
-        # Clang needs it for some atomic operations (https://clang.llvm.org/docs/Toolchain.html#atomics-library).
-        ['(OS=="linux" and clang==1) or (v8_current_cpu in ["mips64", "mips64el", "arm", "riscv64", "loong64"])', {
-          'link_settings': {
-            'libraries': ['-latomic', ],
-          },
-        }],
         ['node_shared_abseil=="false"', {
           'dependencies': ['abseil.gyp:abseil'],
         }],
