diff --git a/setup.py b/setup.py
index 05c2d63..7a5efd2 100644
--- a/setup.py
+++ b/setup.py
@@ -4,7 +4,7 @@
 from distutils.core import setup
 
 packages = \
-['wheel', 'wheel.cli', 'wheel.vendored', 'wheel.vendored.packaging']
+['wheel', 'wheel.cli']
 
 package_data = \
 {'': ['*']}
diff --git a/src/wheel/metadata.py b/src/wheel/metadata.py
index b8098fa..3a17175 100644
--- a/src/wheel/metadata.py
+++ b/src/wheel/metadata.py
@@ -13,7 +13,7 @@ from email.message import Message
 from email.parser import Parser
 from typing import Generator, Iterable, Iterator, Literal
 
-from .vendored.packaging.requirements import Requirement
+from packaging.requirements import Requirement
 
 
 def _nonblank(str: str) -> bool | Literal[""]:
diff --git a/tests/test_bdist_wheel.py b/tests/test_bdist_wheel.py
index bb1457a..6e118ef 100644
--- a/tests/test_bdist_wheel.py
+++ b/tests/test_bdist_wheel.py
@@ -22,7 +22,7 @@ from wheel._bdist_wheel import (
     remove_readonly,
     remove_readonly_exc,
 )
-from wheel.vendored.packaging import tags
+from packaging import tags
 from wheel.wheelfile import WheelFile
 
 DEFAULT_FILES = {
