diff --git a/third_party/libwebrtc/rtc_base/platform_thread_types.cc b/third_party/libwebrtc/rtc_base/platform_thread_types.cc
index 697beda..3da0a9e 100644
--- a/third_party/libwebrtc/rtc_base/platform_thread_types.cc
+++ b/third_party/libwebrtc/rtc_base/platform_thread_types.cc
@@ -12,7 +12,6 @@
 
 // IWYU pragma: begin_keep
 #if defined(WEBRTC_LINUX)
-#include <linux/prctl.h>
 #include <sys/prctl.h>
 #include <sys/syscall.h>
 
