diff --git a/libass/ass_render.h b/libass/ass_render.h
index 7a157ab..c8adf1d 100644
--- a/libass/ass_render.h
+++ b/libass/ass_render.h
@@ -22,7 +22,6 @@
 
 #include <inttypes.h>
 #include <stdbool.h>
-#include <fribidi.h>
 #include <ft2build.h>
 #include FT_FREETYPE_H
 #include FT_GLYPH_H
@@ -49,6 +48,8 @@
 #define PARSED_FADE (1<<0)
 #define PARSED_A    (1<<1)
 
+typedef uint32_t FriBidiChar;
+
 typedef struct {
     ASS_Image result;
     CompositeHashValue *source;
diff --git a/libass/ass_shaper.c b/libass/ass_shaper.c
index 86f2c66..e72a7d5 100644
--- a/libass/ass_shaper.c
+++ b/libass/ass_shaper.c
@@ -87,8 +87,7 @@ struct ass_shaper_metrics_data {
  */
 void ass_shaper_info(ASS_Library *lib)
 {
-    ass_msg(lib, MSGL_INFO, "Shaper: FriBidi "
-            FRIBIDI_VERSION " (SIMPLE)"
+    ass_msg(lib, MSGL_INFO, "Shaper: "
             " HarfBuzz-ng %s (COMPLEX)", hb_version_string()
            );
 }
@@ -705,8 +704,7 @@ static bool shape_harfbuzz(ASS_Shaper *shaper, GlyphInfo *glyphs, size_t len)
                     lead_context, i - offset + 1);
         }
 
-        props.direction = FRIBIDI_LEVEL_IS_RTL(level) ?
-            HB_DIRECTION_RTL : HB_DIRECTION_LTR;
+        props.direction = HB_DIRECTION_LTR;
         props.script = glyphs[offset].script;
         props.language  = hb_shaper_get_run_language(shaper, props.script);
         hb_buffer_set_segment_properties(buf, &props);
@@ -774,35 +772,6 @@ void ass_shaper_determine_script(ASS_Shaper *shaper, GlyphInfo *glyphs,
     }
 }
 
-/**
- * \brief Shape event text with FriBidi. Does mirroring and simple
- * Arabic shaping.
- * \param len number of clusters
- */
-static void shape_fribidi(ASS_Shaper *shaper, GlyphInfo *glyphs, size_t len)
-{
-    int i;
-    FriBidiJoiningType *joins = calloc(len, sizeof(*joins));
-
-    // shape on codepoint level
-    fribidi_get_joining_types(shaper->event_text, len, joins);
-    fribidi_join_arabic(shaper->ctypes, len, shaper->emblevels, joins);
-    fribidi_shape(FRIBIDI_FLAGS_DEFAULT | FRIBIDI_FLAGS_ARABIC,
-            shaper->emblevels, len, joins, shaper->event_text);
-
-    // update indexes
-    for (i = 0; i < len; i++) {
-        GlyphInfo *info = glyphs + i;
-        FT_Face face = info->font->faces[info->face_index];
-        info->symbol = shaper->event_text[i];
-        info->glyph_index = ass_font_index_magic(face, shaper->event_text[i]);
-        if (info->glyph_index)
-            info->glyph_index = FT_Get_Char_Index(face, info->glyph_index);
-    }
-
-    free(joins);
-}
-
 /**
  * \brief Toggle kerning for HarfBuzz shaping.
  * \param shaper shaper instance
@@ -895,7 +864,7 @@ void ass_shaper_set_base_direction(ASS_Shaper *shaper, FriBidiParType dir)
     shaper->base_direction = dir;
 
     if (shaper->whole_text_layout != WHOLE_TEXT_LAYOUT_EXPLICIT)
-        shaper->whole_text_layout = dir == FRIBIDI_PAR_ON ?
+        shaper->whole_text_layout = dir == 0 ?
             WHOLE_TEXT_LAYOUT_IMPLICIT : WHOLE_TEXT_LAYOUT_OFF;
 }
 
@@ -935,7 +904,7 @@ void ass_shaper_set_whole_text_layout(ASS_Shaper *shaper, bool enable)
 {
     shaper->whole_text_layout = enable ?
         WHOLE_TEXT_LAYOUT_EXPLICIT :
-        shaper->base_direction == FRIBIDI_PAR_ON ?
+        shaper->base_direction == 0 ?
             WHOLE_TEXT_LAYOUT_IMPLICIT : WHOLE_TEXT_LAYOUT_OFF;
 }
 
@@ -946,8 +915,7 @@ void ass_shaper_set_whole_text_layout(ASS_Shaper *shaper, bool enable)
  */
 bool ass_shaper_shape(ASS_Shaper *shaper, TextInfo *text_info)
 {
-    int i, ret, last_break;
-    FriBidiParType dir, *pdir;
+    int i;
     GlyphInfo *glyphs = text_info->glyphs;
     shaper->event_text = text_info->event_text;
 
@@ -957,61 +925,7 @@ bool ass_shaper_shape(ASS_Shaper *shaper, TextInfo *text_info)
     for (i = 0; i < text_info->length; i++)
         shaper->event_text[i] = glyphs[i].symbol;
 
-    fribidi_get_bidi_types(shaper->event_text,
-            text_info->length, shaper->ctypes);
-
-    int n_pars = 1;
-    for (i = 0; i < text_info->length - 1; i++)
-        if (shaper->ctypes[i] == FRIBIDI_TYPE_BS)
-            n_pars++;
-
-    if (!check_par_allocations(shaper, n_pars))
-        return false;
-
-#ifdef USE_FRIBIDI_EX_API
-    if (shaper->bidi_brackets) {
-        fribidi_get_bracket_types(shaper->event_text,
-                text_info->length, shaper->ctypes, shaper->btypes);
-    }
-#endif
-
-    // Get bidi embedding levels
-    last_break = 0;
-    pdir = shaper->pbase_dir;
-    for (i = 0; i < text_info->length; i++) {
-        // Embedding levels must be calculated one bidi "paragraph" at a time
-        if (i == text_info->length - 1 ||
-                shaper->ctypes[i] == FRIBIDI_TYPE_BS ||
-                (!shaper->whole_text_layout &&
-                    (glyphs[i + 1].starts_new_run || glyphs[i].hspacing))) {
-            dir = shaper->base_direction;
-#ifdef USE_FRIBIDI_EX_API
-            FriBidiBracketType *btypes = NULL;
-            if (shaper->bidi_brackets)
-                btypes = shaper->btypes + last_break;
-            ret = fribidi_get_par_embedding_levels_ex(
-                    shaper->ctypes + last_break, btypes,
-                    i - last_break + 1, &dir, shaper->emblevels + last_break);
-#else
-            ret = fribidi_get_par_embedding_levels(shaper->ctypes + last_break,
-                    i - last_break + 1, &dir, shaper->emblevels + last_break);
-#endif
-            if (ret == 0)
-                return false;
-            last_break = i + 1;
-            if (shaper->whole_text_layout)
-                *pdir++ = dir;
-        }
-    }
-
-    switch (shaper->shaping_level) {
-    case ASS_SHAPING_SIMPLE:
-        shape_fribidi(shaper, glyphs, text_info->length);
-        return true;
-    case ASS_SHAPING_COMPLEX:
-    default:
-        return shape_harfbuzz(shaper, glyphs, text_info->length);
-    }
+    return shape_harfbuzz(shaper, glyphs, text_info->length);
 }
 
 /**
@@ -1025,7 +939,7 @@ ASS_Shaper *ass_shaper_new(Cache *metrics_cache, Cache *face_size_metrics_cache)
     if (!shaper)
         return NULL;
 
-    shaper->base_direction = FRIBIDI_PAR_ON;
+    shaper->base_direction = 0;
 
     if (!init_features(shaper))
         goto error;
@@ -1087,38 +1001,12 @@ void ass_shaper_cleanup(ASS_Shaper *shaper, TextInfo *text_info)
  */
 FriBidiStrIndex *ass_shaper_reorder(ASS_Shaper *shaper, TextInfo *text_info)
 {
-    int i, ret;
+    int i;
 
     // Initialize reorder map
     for (i = 0; i < text_info->length; i++)
         shaper->cmap[i] = i;
 
-    // Create reorder map line-by-line or run-by-run
-    int last_break = 0;
-    FriBidiParType *pdir = shaper->whole_text_layout ?
-        shaper->pbase_dir : &shaper->base_direction;
-    GlyphInfo *glyphs = text_info->glyphs;
-    for (i = 0; i < text_info->length; i++) {
-        // Bidi "paragraph separators" may occur between line breaks:
-        // U+001C..1E even with ASS_FEATURE_WRAP_UNICODE,
-        // or U+000D, U+0085, U+2029 only without it
-        if (i == text_info->length - 1 || glyphs[i + 1].linebreak ||
-                shaper->ctypes[i] == FRIBIDI_TYPE_BS ||
-                (!shaper->whole_text_layout &&
-                    (glyphs[i + 1].starts_new_run || glyphs[i].hspacing))) {
-            ret = fribidi_reorder_line(0,
-                    shaper->ctypes, i - last_break + 1, last_break, *pdir,
-                    shaper->emblevels, NULL,
-                    shaper->cmap);
-            if (ret == 0)
-                return NULL;
-
-            last_break = i + 1;
-            if (shaper->whole_text_layout && shaper->ctypes[i] == FRIBIDI_TYPE_BS)
-                pdir++;
-        }
-    }
-
     return shaper->cmap;
 }
 
@@ -1136,10 +1024,5 @@ FriBidiStrIndex *ass_shaper_get_reorder_map(ASS_Shaper *shaper)
  */
 FriBidiParType ass_resolve_base_direction(int enc)
 {
-    switch (enc) {
-        case -1:
-            return FRIBIDI_PAR_ON;
-        default:
-            return FRIBIDI_PAR_LTR;
-    }
+    return 0;
 }
diff --git a/libass/ass_shaper.h b/libass/ass_shaper.h
index 06f868d..333252b 100644
--- a/libass/ass_shaper.h
+++ b/libass/ass_shaper.h
@@ -21,14 +21,15 @@
 
 typedef struct ass_shaper ASS_Shaper;
 
-#include <fribidi.h>
 #include <stdbool.h>
 #include "ass_render.h"
 #include "ass_cache.h"
 
-#if FRIBIDI_MAJOR_VERSION >= 1
-#define USE_FRIBIDI_EX_API
-#endif
+typedef uint32_t FriBidiChar;
+typedef uint32_t FriBidiCharType;
+typedef int FriBidiStrIndex;
+typedef int FriBidiParType;
+typedef signed char FriBidiLevel;
 
 void ass_shaper_info(ASS_Library *lib);
 ASS_Shaper *ass_shaper_new(Cache *metrics_cache, Cache *face_size_metrics_cache);
diff --git a/meson.build b/meson.build
index 9eb7969..bc40a57 100644
--- a/meson.build
+++ b/meson.build
@@ -87,12 +87,6 @@ deps += dependency(
     default_options: ['harfbuzz=disabled'],
 )
 
-deps += dependency(
-    'fribidi',
-    version: '>= 0.19.1',
-    default_options: ['docs=false', 'tests=false'],
-)
-
 harfbuzz_options = [
     'tests=disabled',
     'cairo=disabled',
