diff --git a/libavformat/avformat.h b/libavformat/avformat.h
index 1916aa2..43aa2dd 100644
--- a/libavformat/avformat.h
+++ b/libavformat/avformat.h
@@ -1019,6 +1019,8 @@ attribute_deprecated
 int64_t    av_stream_get_end_pts(const AVStream *st);
 #endif
 
+int64_t    av_stream_get_first_dts(const AVStream *st);
+
 #define AV_PROGRAM_RUNNING 1
 
 /**
diff --git a/libavformat/mux_utils.c b/libavformat/mux_utils.c
index 3e63b80..758d985 100644
--- a/libavformat/mux_utils.c
+++ b/libavformat/mux_utils.c
@@ -40,6 +40,11 @@ int64_t av_stream_get_end_pts(const AVStream *st)
 }
 #endif
 
+int64_t av_stream_get_first_dts(const AVStream *st)
+{
+    return cffstream(st)->first_dts;
+}
+
 int avformat_query_codec(const AVOutputFormat *ofmt, enum AVCodecID codec_id,
                          int std_compliance)
 {
