From 979677eb2560122138c5857b4c2b8e387220a0cd Mon Sep 17 00:00:00 2001
Message-Id: <979677eb2560122138c5857b4c2b8e387220a0cd.1672107964.git.owen@owenrafferty.com>
From: Owen Rafferty <owen@owenrafferty.com>
Date: Mon, 26 Dec 2022 20:20:01 -0600
Subject: [PATCH] only build support for elf amd64

---
 CMakeLists.txt | 12 +-----------
 elf/main.cc    |  5 -----
 2 files changed, 1 insertion(+), 16 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b985719..0779117c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -275,8 +275,7 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 # compiler instances. This is hacky but greatly reduces compile time
 # on a multicore machine.
 list(APPEND MOLD_ELF_TARGETS
-  X86_64 I386 ARM64 ARM32 RV32LE RV32BE RV64LE RV64BE
-  PPC64V1 PPC64V2 S390X SPARC64 M68K)
+  X86_64)
 
 list(APPEND MOLD_ELF_TEMPLATE_FILES
   elf/cmdline.cc
@@ -337,15 +336,6 @@ endif()
 target_sources(mold PRIVATE
   compress.cc
   demangle.cc
-  elf/arch-arm32.cc
-  elf/arch-arm64.cc
-  elf/arch-i386.cc
-  elf/arch-m68k.cc
-  elf/arch-ppc64v1.cc
-  elf/arch-ppc64v2.cc
-  elf/arch-riscv.cc
-  elf/arch-s390x.cc
-  elf/arch-sparc64.cc
   elf/arch-x86-64.cc
   filepath.cc
   git-hash.cc
diff --git a/elf/main.cc b/elf/main.cc
index ed5b3c03..8bd193f1 100644
--- a/elf/main.cc
+++ b/elf/main.cc
@@ -381,11 +381,6 @@ int elf_main(int argc, char **argv) {
   if (ctx.arg.emulation == MachineType::NONE)
     ctx.arg.emulation = deduce_machine_type(ctx, file_args);
 
-  // Redo if -m is not x86-64.
-  if constexpr (std::is_same_v<E, X86_64>)
-    if (ctx.arg.emulation != MachineType::X86_64)
-      return redo_main<E>(argc, argv, ctx.arg.emulation);
-
   Timer t_all(ctx, "all");
 
   install_signal_handler();
-- 
2.39.0

